File: 7-kyu/reverse-words/c++/solution.cpp
def reverse_words(text):
    ret = ''
    i = 0
    while i < len(text):
        if text[i].isspace():
            spc = ''
            while i < len(text) and text[i].isspace():
                spc += text[i]
                i += 1
            ret += spc
        else:
            wrd = ''
            while i < len(text) and not text[i].isspace():
                wrd += text[i]
                i += 1
            wrd = wrd[::-1]
            ret += wrd
    return ret

File: 7-kyu/reverse-words/javascript/solution.js
#include <algorithm>
#include <cctype>

std::string reverse_words(std::string str)
{
    std::string ret = "";
    if (str.empty()) return ret;
    size_t i = 0;
    while (i < str.length()) {
        if (std::isspace(str[i])) {
            std::string spc = "";
            while (i < str.length() && std::isspace(str[i])) {
                spc += str[i];
                i++;
            }
            ret += spc;
        } else {
            std::string wrd = "";
            while (i < str.length() && !(std::isspace(str[i]))) {
                wrd += str[i];
                i++;
            }
            std::reverse(wrd.begin(), wrd.end());
            ret += wrd;
        }
    }
    return ret;
}